From e5db8238d6118d9ebdc9e97e71c70fe0b6a338fa Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Mon, 3 Aug 2026 15:01:27 -0400
Subject: [PATCH 256/268] HACK: winewayland: Add env to enable IME.

Turns out even with all the IME improvements the IME implementations in games are total horse shit, so lets just turn it off for now.
---
 dlls/winewayland.drv/wayland.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/dlls/winewayland.drv/wayland.c b/dlls/winewayland.drv/wayland.c
index 28829b7150d..ad36bd20773 100644
--- a/dlls/winewayland.drv/wayland.c
+++ b/dlls/winewayland.drv/wayland.c
@@ -146,6 +146,17 @@ static int wayland_disable_ssd(void)
     return disabled;
 }
 
+static int wayland_use_ime(void)
+{
+    static int enabled = -1;
+    const char *env;
+
+    if (enabled == -1)
+        enabled = (env = getenv("WAYLANDDRV_IME")) && !strcmp(env, "1");
+
+    return enabled;
+}
+
 /**********************************************************************
  *          Registry handling
  */
@@ -235,6 +246,7 @@ static void registry_handle_global(void *data, struct wl_registry *registry,
     }
     else if (strcmp(interface, "zwp_text_input_manager_v3") == 0)
     {
+        if (!wayland_use_ime()) return;
         process_wayland.zwp_text_input_manager_v3 =
             wl_registry_bind(registry, id, &zwp_text_input_manager_v3_interface, 1);
         if (process_wayland.seat.wl_seat) wayland_text_input_init();
-- 
2.55.0

